Analysis/AnalysisCacheProducer.py

```diff
@@ -54,6 +56,27 @@ def createCentralQuantities(df_central, central_col_types, central_columns):
     df_central = map_creator.processCentral(ROOT.RDF.AsRNode(df_central), Utilities.ListToVector(central_columns))
     return df_central
 
+def run_producer(producer, dfw, producer_config, outFileName, treeName):
+    if producer_config.get('awkward_based', False):
+        dfw = producer.prepare_dfw(dfw)
+        vars_to_save = producer.vars_to_save
+        dfw.df.Snapshot(f'tmp', f'tmp.root', vars_to_save, snapshotOptions)
+        final_array = None
+        for array in uproot.iterate('tmp.root:tmp', step_size='50MB'): # For DNN, this translates to ~300_000 events
+            new_array = producer.run(array)
+            if final_array is None:
+                final_array = new_array
+            else:
+                final_array = ak.concatenate([final_array, new_array])
+        with uproot.recreate(outFileName) as outfile:
+            outfile[treeName] = final_array
+
+    else:
+        dfw = producer.run(dfw)
+        varToSave = Utilities.ListToVector(dfw.colToSave)
```

Comment:
Check that `dfw.colToSave` contains only variables declared in the config + `FullEventId`